### PR TITLE
Now compatible with Unicode astral characters and emoji

### PR DESCRIPTION
**Pull request by revelt**
The updated algorithm is not as elegant as original (splits the characters into array before slicing) but it does recognise Unicode's astral characters and emoji correctly now.

### CODE REVIEW

**Comment by sindresorhus** [comment]
This seems to still fail with your pull request:

```
t.is(fn('❤️🐴🐴', 1, 1, '🦄'), '❤️🦄🐴');
```

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "splice-string",
-  "version": "1.0.0",
+  "version": "1.1.1",
```
Comment (sindresorhus) [inline package.json]: Don't bump the version.

**Comment by revelt** [comment]
Good spot! Back to coding then.

**Comment by revelt** [comment]
Apparently, Node `0.10` is not emoji-friendly, while `0.12`-onwards is: https://tonicdev.com/57ac7398a55ecf1200fa1a05/57aceab067ea8c1300c1a1d1
If you flip the Tonicdev link's setting on the dropbox it does work on all other versions.

So it's `.travis.yml` setting asking for `0.10`.

**Comment by sindresorhus** [comment]
My provided test case fails on Node.js 4 too.

I plan on dropping Node.js 0.10 and 0.12 after this PR btw.

**Comment by revelt** [comment]
Hi Sindre @sindresorhus 

What if we just forgot this PR and you did what you think it's best to support astral chars?
Feel free to reuse anything from my PR. No need to bother with attribution etc.
I just terribly need splice with astral support (like yesterday), it's blocking my next big idea and I trust your capable hands 😄 

What do you think?

cheers
Roy

**Comment by sindresorhus** [comment]
Fixed in 2.0.0.

**Pull request closed without merging**